lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/driver/RuntimesHost.java

```diff
@@ -127,6 +131,17 @@ private void initializeInjector(final Injector runtimeInjector) throws Injection
     runtimeInjector.bindVolatileParameter(
             RuntimeParameters.RuntimeStatusHandler.class,
             runtimeStatusEventHandler);
+    HttpServer httpServer = null;
+    try {
+      httpServer = this.originalInjector.getInstance(HttpServer.class);
+    } catch(InjectionException e) {
+      LOG.log(Level.INFO, "Http Server is not configured for the runtime");
```

Comment:
This is not a warning, as http server is not a must. So the exception is normal here. I do that as we don;t have any other way to test if an implementation exists.